blog_article/blog_article.adoc

```diff
@@ -0,0 +1,161 @@
+that are created with HTTP requests are not client-bound and therefore cannot be created as "exclusive". Automatic
+deletion needs to be achieved through other queue properties like _auto-delete_ and _TTL_. In addition, the REST API does
+not offer to create queues without names to let the broker choose a (temporary) unique queue name. This leads to the
+consequence that unique names have to be managed by the application.  +
+In most cases, these requirements will only be
+needed in specific scenarios. Nevertheless, it should be clear that RabbitMQ in conjunction with AMQP 1.0 can only be
+used with partially limited functionality.
+
+== AMQP 0.9.1 and AMQP 1.0 Interoperability
+As we already know, AMQP 0.9.1 and AMQP 1.0 should be considered as completely different specifications which leads
+to the questions: Is interoperability between these two protocols possible
+"out-of-the-box"? And if not, what has to be considered to achieve interoperability? What about interoperability
+between different clients speaking the same protocol? +
+These points are especially important when we think again about an (evolving) event-driven architecture. The components
+of a system should be able to exchange events and data independently of the protocol they are using. +
+As also described within RabbitMQ’s plugin documentation, interoperability between AMQP 0.9.1 and AMQP 1.0 is provided
+with some constraints:
+
+|===
+|Producer Library |Consumer Library|Interoperability
+
+|SwiftMQ 1.0
+|RabbitMQ 0.9.1
+|provided with constraints -> sending as plain bytes
+
+|RabbitMQ 0.9.1
+|SwiftMQ 1.0
+|provided with constraints -> reading from data field
+
+|Qpid JMS 1.0
+|SwiftMQ 1.0
+|provided
+
+|SwiftMQ 1.0
+|Qpid JMS 1.0
+|provided
+
+|Qpid JMS 1.0
+|RabbitMQ 0.9.1
+|provided with constraints -> sending as plain bytes
+
+|RabbitMQ 0.9.1
+|Qpid JMS 1.0
+|provided
+|===
+
+Like denoted in the table above, when testing the different clients, it was generally possible to send and receive
+messages using AMQP 1.0 and AMQP 0.9.1. +
+For AMQP 1.0 messages that should be read by AMQP 0.9.1 clients, it should be considered that AMQP 1.0 uses
+the so-called AMQP-encoding. This means that all messages that are sent with AMQP 1.0 are encoded beforehand.
+The grammar of this encoding is specified within the AMQP 1.0 protocol. However, this grammar can be implemented
+differently by each client. +
+The consequences will become clear when looking at an example:
+
+* Sent message using SwiftMQ client (AMQP 1.0): M1
+* Received message using RabbitMQ client (AMQP 0.9.1): �w�M1
+
+* Sent message using Qpid JMS client (AMQP 1.0): M1
+* Received message using RabbitMQ client (AMQP 0.9.1): Sw�M1
+
+The examples show that there are some additional bytes appended in front of the messages. The reason for this is the
+AMQP-encoding that is used by the message producers. Since this encoding seems not to be implemented the same for
```

Comment:
Both follow the AMQP spec:

Qpid JMS (M1 as Amqp-Value): 0x00 0x53 0x77 0xA1 0x02 M1
--> 0x00 0x53 -> Descriptor type described by smallulong code (1 Byte)
--> 0x77 -> Amqp Value Described Type
--> 0xA1 -> String
--> 0x02 -> Size of String

SwiftMQ (M1 as Amqp-Value): 0x00 0x80 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x77 0xA1 0x02 M1
--> 0x00 0x80 -> Descriptor type described by ulong code (8 Bytes)
--> 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x77 -> Amqp Value Described Type
--> 0xA1 -> String
--> 0x02 -> Size of String

So both exactly follow the specification of AMQP 1.0 type system (http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#section-types). Thats the reason, why the can exchange messages, because both actually mean the same. Qpid JMS and SwiftMQ just use a slightly different primitive data type for encoding the described type. Qpid uses 0x53 (small unsigned long) which is basically totally enough and SwiftMq uses 0x80 (a unsiged long with 8 bytes). Of course, Qpid JMS is much more efficient in sense of transferred bytes, however both is totally valid and works.

Of course one may ask if it is a good approach to allow so many different ways of saying the same thing. In my opinion its not. I personally think the Amqp 1.0 type system us much to complex and it would be much easier to just use Avro for example. The specification states that the type system is made for interoperable data representation (http://docs.oasis-open.org/amqp/core/v1.0/os/amqp-core-types-v1.0-os.html#section-types). However, if someone wants to have a interoperable data format I would prefere Json, Avro or Protobuf, because today the are more widely used.
This could also be a conclusion of this blog post. That in our opinion AMQP 1.0 data format / type system should not be used, because its actually not interoperable.